src/hyperion/experiment_plans/robot_load_then_centre_plan.py

```diff
@@ -146,6 +149,36 @@ def prepare_for_robot_load(composite: RobotLoadThenCentreComposite):
     yield from bps.wait("prepare_robot_load")
 
 
+def robot_load_and_energy_change(
+    composite: RobotLoadThenCentreComposite,
+    sample_location: SampleLocation,
+    demand_energy_ev: float | None,
+):
+    yield from bps.abs_set(
+        composite.robot,
+        sample_location,
+        group="robot_load",
+    )
+
+    if demand_energy_ev:
+        yield from set_energy_plan(
```

Comment:
bit tangential, but does the set energy plan always have to block during robot load or could we wait on it later?

Reply:
No, we want https://github.com/DiamondLightSource/hyperion/issues/1495 really but needs the RE changes